client/my-sites/comments/comment/comment-actions.jsx

```diff
@@ -93,6 +101,64 @@ export class CommentActions extends Component {
 		removeFromPersisted( commentId );
 	};
 
+	showNotice = status => {
+		const {
+			changeStatus,
+			commentId,
+			like,
+			minimumComment,
+			postId,
+			removeFromPersisted,
+			siteId,
+			translate,
+			unlike,
+			updateLastUndo,
+		} = this.props;
+
+		this.props.removeNotice( 'comment-notice' );
+
+		const message = get(
+			{
+				approved: translate( 'Comment approved.' ),
+				unapproved: translate( 'Comment unapproved.' ),
+				spam: translate( 'Comment marked as spam.' ),
+				trash: translate( 'Comment moved to trash.' ),
+			},
+			status
+		);
+
+		const noticeOptions = {
+			button: translate( 'Undo' ),
+			id: 'comment-notice',
+			isPersistent: true,
+			onClick: () => {
```

Comment:
Is it possible pulling this up, or did we need the closure here? Looks like most of this should be available from props.